hal/src/nRF52840/ble_hal.cpp

```diff
@@ -15,739 +15,3147 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "logging.h"
+LOG_SOURCE_CATEGORY("hal.ble")
+
 #include "ble_hal.h"
 
-#include "nrf_ble_gatt.h"
-#include "ble_advertising.h"
-#include "ble_conn_params.h"
-#include "ble.h"
+#if HAL_PLATFORM_BLE
 
+/* Headers included from nRF5_SDK/components/softdevice/s140/headers */
+#include "ble.h"
+#include "ble_gap.h"
+#include "ble_gatt.h"
+#include "ble_gatts.h"
+#include "ble_gattc.h"
+#include "ble_types.h"
+/* Headers included from nRF5_SDK/components/softdevice/common */
 #include "nrf_sdh.h"
 #include "nrf_sdh_ble.h"
 #include "nrf_sdh_soc.h"
 
-#include "logging.h"
+#include "app_util.h"
+
+#include "concurrent_hal.h"
+#include "static_recursive_mutex.h"
+#include <mutex>
+
+#include "gpio_hal.h"
+#include "device_code.h"
+#include "system_error.h"
+#include "sdk_config_system.h"
+#include "spark_wiring_vector.h"
+#include "simple_pool_allocator.h"
+#include <string.h>
+#include <memory>
+
+static_assert(NRF_SDH_BLE_PERIPHERAL_LINK_COUNT == 1, "Multiple simultaneous peripheral connections are not supported");
+static_assert(NRF_SDH_BLE_TOTAL_LINK_COUNT <= 20, "Maximum supported number of concurrent connections in the peripheral and central roles combined exceeded");
+
+using spark::Vector;
+
+#define BLE_CONN_CFG_TAG 1
+
+#define NRF_CHECK_RETURN(ret, str) \
```

Comment:
Let's use macros from `check.h` and `check_nrf.h` instead of redefining them here.